extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -456,15 +464,124 @@ private static Map<String, ParquetTableWriter.GroupingColumnWritingInfo> groupin
         return gcwim;
     }
 
+    /**
+     * Write table to disk in parquet format with {@link TableDefinition#getPartitioningColumns() partitioning columns}
+     * written as "key=value" format in a nested directory structure. To generate these individual partitions, this
+     * method will call {@link Table#partitionBy(String...) partitionBy} on all the partitioning columns.
+     *
+     * @param sourceTable The table to partition and write
+     * @param destinationDir The destination directory to store partitioned data in. Non-existing directories are
+     *        created.
+     * @param baseName The base name for the individual partitioned tables. For example, a base name of "table" will
+     *        result in files named "partition1/table.parquet", "partition2/table.parquet", etc.
+     * @param writeInstructions Write instructions for customizations while writing
+     */
+    public static void writeKeyValuePartitionedTable(@NotNull final Table sourceTable,
+            @NotNull final File destinationDir,
+            @NotNull final String baseName,
+            @NotNull final ParquetInstructions writeInstructions) {
+        // TODO Should I take an additional write instruction for partitioning column names in case they are different
+        // from the table's partitioning columns? Also, should I take partitioning column names from the user as an
+        // argument to this method?
+
+        // Also, how should I get the baseName, should I move it to the write instructions.
+        // Pyarrow has this optional parameter "basename_template" used to generate basenames of written data files.
+        // The token ‘{i}’ will be replaced with an automatically incremented integer for files in the same folder.
+        // If not specified, it defaults to “someHash-{i}.parquet”.
+        // pyspark has names of the form "part.{i}.parquet" and allows passing a naming function which takes an integer
+        // and generates a file name.
+        final List<ColumnDefinition<?>> partitioningColumns = sourceTable.getDefinition().getPartitioningColumns();
+        if (partitioningColumns.isEmpty()) {
+            throw new IllegalArgumentException("Table must have partitioning columns to write partitioned data");
+        }
+        final String[] partitioningColNames = partitioningColumns.stream()
+                .map(ColumnDefinition::getName)
+                .toArray(String[]::new);
+        final PartitionedTable partitionedTable = sourceTable.partitionBy(partitioningColNames);
```

Comment:
Yes, keeping keys and not adding them to metadata.
So there's not much advantage of doing "drop keys" because in the other api which takes a partitioned table, we cannot assume that the user has dropped the keys.